lib/spack/spack/bootstrap/core.py

```diff
@@ -559,12 +559,47 @@ def ensure_patchelf_in_path_or_raise() -> spack.util.executable.Executable:
         )
 
 
+def ensure_winsdk_or_raise() -> None:
+    """Ensure the Windows SDK + WGL are available on system
+    If both of these package are found, the Spack user or bootstrap
+    configuration (depending on where Spack is running)
+    will be updated to include all versions and variants detected.
+    If either the WDK or WSDK are not found, this method will raise
+    a RuntimeError.
+
+    **NOTE:** This modifies the Spack config in the current scope,
+    either user or environment depending on the calling context.
+    This is different from all other current bootstrap dependency
+    checks.
+    """
+    externals = spack.detection.by_path(["win-sdk", "wgl"])
```

Comment:
I just realized: we probably also want to initially check the config to see if we already have these.

To that end, a slightly clearer method name might be `ensure_winsdk_external_or_raise`.